pennylane_lightning/src/rework/StateVector.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2021 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "StateVector.hpp"
+
+Pennylane::StateVector Pennylane::StateVector::create(const pybind11::array_t<CplxType>* numpyArray) {
```

Comment:
Curious about merging `create` into the constructor here too 🤔

Reply:
This was a somewhat intentional choice--we don't really need a `pybind::array_t<CplxType>` in order to initialise what's really just a wrapper around a `CplxType*` array (especially, say, if we wanted to unit test this). Farming it out into a `create` method allows the validation checks to be done prior to class instantiation.